### DISCUSSION TITLE
`Sms` control - send and receive SMS

### DISCUSSION THREAD

**Discussion by bitmvk** [flet-dev/flet#4900] [Extensions]
**Please Describe The Feature To Be Added**

SMS and MMS reading and sending abilities

**Suggest A Solution**
https://pub.dev/packages/flutter_sms

**Comment by hanchenglong001** [comment]
Can you tell me how to integrate the existing Flutter package into my Flet application? 
No examples were found on the official website.

**Comment by bitmvk** [comment]
hey @hanchenglong001,

Did you see the blog post called [Flet adaptive UI and custom controls release](https://flet.dev/blog/flet-adaptive-and-custom-controls#integrating-existing-flutter-packages)?
You can see the implementation of all [packages](https://github.com/flet-dev/flet/tree/main/packages) too!

After creating the dart wrapper code, you have to edit the [client/lib/main.dart](https://github.com/flet-dev/flet/blob/main/client/lib/main.dart) and the [client/pubspec.yaml](https://github.com/flet-dev/flet/blob/main/client/pubspec.yaml) files. Just follow as done with other packages.

On the python side, just create a new file in [flet-core/src/flet-core](https://github.com/flet-dev/flet/tree/main/sdk/python/packages/flet-core/src/flet_core) and import the Control class created into the `__init__.py` file. Just follow as done in other controls like `audio.py`, `video.py` ,`rive.py`, etc...

Once finished, you can test it out as told in [contributing docs](https://github.com/flet-dev/flet/blob/main/CONTRIBUTING.md)

**Note**: this is just for now the moderators seem to be trying to make  integration of flutter packages to flet easier, so there might be changes on how to do it.
